packages/squiggle-lang/src/ast/peggyParser.peggy

```diff
@@ -50,19 +51,26 @@ quotedInnerBlock
     '}'
     { return h.nodeBlock([finalExpression], location()); }
 
-statementsList = statement|1.., statementSeparator|
+statementsList = decoratedStatement|1.., statementSeparator|
 
-statement
-  = letStatement
-  / defunStatement
-  / voidStatement
-
-voidStatement
-  = "call" _nl value:zeroOMoreArgumentsBlockOrExpression
+decoratedStatement
+  = decorator:decorator statement:decoratedStatement
     {
-      const variable = h.nodeIdentifier("_", location());
-      return h.nodeLetStatement(variable, value, false, location());
+      return h.nodeDecoratedStatement(decorator, statement, location())
     }
+  / letStatement
+  / defunStatement
+
+  decorator
+    = '@' name:dollarIdentifier args:(_ '(' _nl @functionArguments _nl ')')?  __nl
```

Comment:
`dollarIdentifier` means that `@foo` is allowed but `@Foo.bar` is not. We can always add it, but for now `expression/compile.ts` relies on identifiers being plain names (because it appends `Tag.` prefix to them).